hnn_core/network.py

```diff
@@ -219,47 +298,436 @@ def __repr__(self):
-
-        # COMMON FEEDS
-        self.pos_dict['common'] = [origin for i in range(self._n_common_feeds)]
-
-        # UNIQUE FEEDS
-        for key in self._p_unique.keys():
-            # create the pos_dict for all the sources
-            self.pos_dict[key] = [origin for i in range(self.n_cells)]
-
-        # Now add the names of the feeds to a list
-        self.feedname_list.append('common')
-        # grab the keys for the unique set of inputs and sort the names
-        # append them to the src list along with the number of cells
-        unique_keys = sorted(self._p_unique.keys())
-        self.feedname_list += unique_keys
+        return deepcopy(self)
 
+    def add_evoked_drive(self, name, *, mu, sigma, numspikes, location,
+                         weights_ampa=None, weights_nmda=None,
+                         space_constant=3., synaptic_delays=0.1, seedcore=2):
+        """Add an 'evoked' external drive to the network
+
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        mu : float
+            Mean of Gaussian event time distribution (across target cells)
+        sigma : float
+            Standard deviation of event time distribution (across target cells)
+            Set to zero for "synchronous" evoked drive.
+        numspikes : int
+            Number of spikes at each target cell
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        weights_ampa : dict or None
+            Synaptic weights (in uS) of AMPA receptors on each targeted cell
+            type (dict keys). Cell types omitted from the dict are set to zero.
+        weights_nmda : dict or None
+            Synaptic weights (in uS) of NMDA receptors on each targeted cell
+            type (dict keys). Cell types omitted from the dict are set to zero.
+        synaptic_delays : float or dict
+            Synaptic delay (in ms) at the column origin, dispersed laterally as
+            a function of the space_constant. If float, applies to all target
+            cell types. Use dict to create delay->cell mapping.
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            weights and delays within the simulated column
+        seedcore : int
+            Optional initial seed for random number generator (default: 2).
+        """
+        self._add_evoked_or_gaussian_drive(
+            'evoked', name, mu, sigma, numspikes, weights_ampa, weights_nmda,
+            location, space_constant=space_constant,
+            synaptic_delays=synaptic_delays, seedcore=seedcore)
+
+    def add_gaussian_drive(self, name, *, mu, sigma, numspikes, location,
+                           weights_ampa=None, weights_nmda=None,
+                           space_constant=100., synaptic_delays=0.1,
+                           seedcore=2):
+        """Add an 'evoked' external drive to the network
```

Comment:
evoked -> Gaussian?

Reply:
Well we already have Gaussian in another context. I don't like it, but it follows GUI variable/method naming. We could consider renaming, but I'd bring in the Jones lab for that discussion.

Reply:
hum ... not sure I follow.

But I just noticed that you have two methods with exactly the same function signature??? I think we should then expose only one, it will get rid of the superfluous nesting. The rest is documentation + fixing how the params are extracted from the param file.

Reply:
If I recall, the only historical difference between what we called a 'gaussian' drive and an 'evoked' drive is that the former was composed of 50 spikes randomly distributed in time rather than just 1. Is that correct?

I don't recall seeing 'gaussian' drives exposed to the user in the GUI, so we're probably safe to remove the distinction between 'gaussian' and 'evoked' drives since the new API makes the number of spikes/input explicit.

Reply:
1. my bad, copy-pasted the docstring!
2. +1 for removing the distinction ('Gaussian' drives are not mentioned in the reference paper)

@rythorpe not sure about the 50 spikes, can't see any evidence of that being hard-coded in the feeds-module. Anyhoot, 50 'evoked' spikes is no problem, though again we have to think about what to do when `t<0` and `t>T`.

Reply:
Removed `add_gaussian_drive` in bab3ff3